src/Build/Logging/FancyLogger/FancyLoggerBuffer.cs

```diff
@@ -0,0 +1,262 @@
+                        case ConsoleKey.Escape:
+                            ToggleAutoScroll();
+                            break;
+                    }
+                }
+            });
+            // Switch to alternate
+            Console.Write(ANSIBuilder.Buffer.UseMainBuffer());
+            Console.Write(ANSIBuilder.Buffer.UseAlternateBuffer());
+            // Settings
+            AutoScrollEnabled = true;
+            // Render contents
+            WriteTitleBar();
+            WriteFooter("This is an empty footer haha");
+            ScrollToEnd();
+        }
+        public static void Terminate()
+        {
+            // Switch to main buffer
+            Console.Write(ANSIBuilder.Buffer.UseMainBuffer());
+            // Dispose event listeners
+            IsTerminated = true;
+            // Delete lines
+            lines = new();
+        }
+
+        #region Scrolling
+        private static void ScrollToLine(int firstLineIndex)
+        {
+            if (firstLineIndex < 0 || firstLineIndex >= lines.Count) return;
+            CurrentTopLineIndex = firstLineIndex;
+            int i = 0;
+            while (i < Height - 4)
+            {
+                int lineIndex = i + firstLineIndex;
+                Console.Write(""
+                    + ANSIBuilder.Cursor.Position(i + 2, 0)
+                    + ANSIBuilder.Eraser.LineCursorToEnd()
+                    // + ((lineIndex < lines.Count && lines[lineIndex].IsHidden) ? " Hidden" : "")
+                    + ((lineIndex < lines.Count) ? ANSIBuilder.Tabulator.ForwardTab(lines[lineIndex].IdentationLevel) + lines[lineIndex].Text : "")
+                );
+                i++;
+            }
+
+            Console.Write(ANSIBuilder.Cursor.Position(Height, 0));
+        }
+        private static void ScrollToEnd()
+        { 
+            // If number of lines is smaller than height
+            if (lines.Count < Height - 2)
+            {
+                ScrollToLine(0);
+            }
+            else
+            {
+                ScrollToLine(lines.Count - Height + 4);
+            }
+            // Go to end
+            Console.Write(ANSIBuilder.Cursor.Position(Height, 0));
+        }
+        private static void ToggleAutoScroll()
+        {
+            AutoScrollEnabled = !AutoScrollEnabled;
+        }
+        #endregion
+        #region Line Referencing
+        public static int GetLineIndexById(int lineId)
+        {
+            for (int i = 0; i < lines.Count; i++)
+            {
+                if (lines[i].Id == lineId) return i;
+            }
+            return -1;
+        }
+        public static FancyLoggerBufferLine? GetLineById(int lineId)
+        {
+            int i = GetLineIndexById(lineId);
+            if (i == -1) return null;
+            return lines[i];
+        }
+        #endregion
+        #region Writing
+        public static void WriteTitleBar()
+        {
+            Console.Write(""
```

Comment:
Why do you have to start there with "" + ? Could you just start with ANSIBuilder.Cursor.Home()?